plugins/hypervisors/kvm/src/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -1764,9 +1765,12 @@ protected ExecutionResult cleanupNetworkElementCommand(final IpAssocCommand cmd)
                 }
                 nicNum = broadcastUriAllocatedToVM.get(ip.getBroadcastUri());
 
-                if (numOfIps == 1 && !ip.isAdd()) {
-                    vifHotUnPlug(conn, routerName, ip.getVifMacAddress());
-                    networkUsage(routerIp, "deleteVif", "eth" + nicNum);
+                if (lastIp != null && lastIp.equalsIgnoreCase("true") && !ip.isAdd()) {
```

Comment:
In CitrixresourceBase StringUtils is used from the  com.cloud.utils.StringUtils. So using  StringUtils from java.lang will be ambiguous.